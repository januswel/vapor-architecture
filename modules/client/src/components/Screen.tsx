import React from 'react'
import { SafeAreaView, StyleSheet, Text } from 'react-native'

import Charger from './Charger'
import Inventory from './Inventory'

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    backgroundColor: '#F5FCFF',
  },
})

export default props => (
  <SafeAreaView style={styles.container}>
    <Charger {...props} />
    <Inventory {...props} />
  </SafeAreaView>
)
