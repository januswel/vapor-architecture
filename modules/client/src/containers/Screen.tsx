import { connect } from 'react-redux'

import { charge, buy } from '../modules/vending-machine'
import Screen from '../components/Screen'

const mapStateToProps = state => ({
  vendingMachine: state,
})

const mapDispatchToProps = dispatch => ({
  actions: {
    charge: chargedMoney => {
      dispatch(charge(chargedMoney))
    },
    buy: selectedIndex => {
      dispatch(buy(selectedIndex))
    },
  },
})

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Screen)
