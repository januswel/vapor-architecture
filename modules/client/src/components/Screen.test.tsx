import * as React from 'react'
import * as renderer from 'react-test-renderer'

import Screen from './Screen'

describe('Screen', () => {
  it('renders screen', () => {
    const component = renderer.create(
      <Screen
        items={[
          {
            id: 0,
            price: 430,
            imageUrl: 'https://example.com/assets/images/coffee.png',
            isSoldOut: false,
            isTooFewChargedMoney: false,
            actions: {
              sellAndCount: () => {},
            },
          },
        ]}
        chargedMoney={300}
        isCommunicating
        actions={{ charge: () => {}, sellAndCount: () => {} }}
      />,
    )
    expect(component).toMatchSnapshot()
  })
})
