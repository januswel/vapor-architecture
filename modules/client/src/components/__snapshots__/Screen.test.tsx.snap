// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen renders screen 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "backgroundColor": "black",
        "color": "orange",
        "fontSize": 32,
        "margin": 8,
        "padding": 16,
        "textAlign": "right",
      }
    }
  >
    ¥ 
    300
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "silver",
            "borderRadius": 48,
            "height": 96,
            "justifyContent": "center",
            "width": 96,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          10
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "silver",
            "borderRadius": 48,
            "height": 96,
            "justifyContent": "center",
            "width": 96,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          50
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "silver",
            "borderRadius": 48,
            "height": 96,
            "justifyContent": "center",
            "width": 96,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          100
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "silver",
            "borderRadius": 48,
            "height": 96,
            "justifyContent": "center",
            "width": 96,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          500
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    data={
      Array [
        Object {
          "actions": Object {
            "sellAndCount": [Function],
          },
          "id": 0,
          "imageUrl": "https://example.com/assets/images/coffee.png",
          "isSoldOut": false,
          "isTooFewChargedMoney": false,
          "price": 430,
        },
      ]
    }
    disableVirtualization={false}
    extraData={300}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={2}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "marginTop": 16,
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "height": 168,
                    "justifyContent": "space-around",
                    "margin": 16,
                    "width": 168,
                  },
                  Object {},
                ]
              }
            >
              <Image
                resizeMode="contain"
                source={
                  Object {
                    "uri": "https://example.com/assets/images/coffee.png",
                  }
                }
                style={
                  Object {
                    "height": 96,
                    "width": 96,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "fontSize": 16,
                  }
                }
              >
                ¥
                430
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "bottom": 0,
        "position": "absolute",
        "right": 0,
      }
    }
  />
</RCTSafeAreaView>
`;
